packages/warriorjs-abilities/src/look.js

```diff
@@ -7,7 +7,14 @@ function look({ range }) {
     description: `Returns an array of up to ${range} spaces in the given direction (${defaultDirection} by default).`,
     perform(direction = defaultDirection) {
       const offsets = Array.from(new Array(range), (_, index) => index + 1);
-      return offsets.map(offset => unit.getSensedSpaceAt(direction, offset));
+      const sensedSpaces = [];
+      offsets
+        .map(offset => unit.getSensedSpaceAt(direction, offset))
+        .some(space => {
+          sensedSpaces.push(space);
+          return space && space.isWall && space.isWall();
```

Comment:
We can safely assume any space returned by `unit.getSensedSpaceAt()` will have the `isWall` method so this check should be simplified.

Reply:
That would be true but the fakes given by the existing unit tests are not of type `Space` so do not have `.isWall` method.

Reply:
Right, you should add `isWall()` to the mocks though now that a simple string does not suffice.

```js
const space1 = { isWall: () => false };
const space2 = { isWall: () => false };
...
unit.getSensedSpaceAt
  .mockReturnValueOnce(space1)
  .mockReturnValueOnce(space2)
...
```